controllers/semeru_compiler.go

```diff
@@ -0,0 +1,206 @@
+		Namespace: wlva.GetNamespace(),
+	}
+	// Deployment
+	semeruDeployment := &appsv1.Deployment{ObjectMeta: compilerMeta}
+	err := r.CreateOrUpdate(semeruDeployment, wlva, func() error {
+		r.reconcileSemeruDeployment(wlva, semeruDeployment)
+		return nil
+	})
+	if err != nil {
+		return err, "Failed to reconcile Deployment : " + semeruDeployment.Name
+	}
+	// Create the Semeru Service object
+	semsvc := &corev1.Service{ObjectMeta: compilerMeta}
+	err = r.CreateOrUpdate(semsvc, wlva, func() error {
+		reconcileSemeruService(semsvc, wlva)
+		return nil
+	})
+	if err != nil {
+		return err, "Failed to reconcile the Semeru Compiler Service"
+	}
+	return nil, ""
+}
+
+func createSemeruService(svc *corev1.Service, la *wlv1.WebSphereLibertyApplication) {
+	var port int32 = 38400
+	var timeout int32 = 86400
+	svc.Labels = getLabels(la)
+	svc.Spec.Selector = getSelectors(la)
+	if len(svc.Spec.Ports) == 0 {
+		svc.Spec.Ports = append(svc.Spec.Ports, corev1.ServicePort{})
+	}
+	svc.Spec.Ports[0].Protocol = corev1.ProtocolTCP
+	svc.Spec.Ports[0].Port = port
+	svc.Spec.Ports[0].TargetPort = intstr.FromInt(int(port))
+	svc.Spec.SessionAffinity = corev1.ServiceAffinityClientIP
+	svc.Spec.SessionAffinityConfig = &corev1.SessionAffinityConfig{
+		ClientIP: &corev1.ClientIPConfig{
+			TimeoutSeconds: &timeout,
+		},
+	}
+}
+
+// Create the Selector map for a Semeru Compiler
+func getSelectors(wlva *wlv1.WebSphereLibertyApplication) map[string]string {
+	requiredSelector := make(map[string]string)
+	requiredSelector["app.kubernetes.io/component"] = SemeruLabelName
+	requiredSelector["app.kubernetes.io/instance"] = wlva.GetName() + SemeruLabelNameSuffix
+	requiredSelector["app.kubernetes.io/part-of"] = wlva.GetName()
+	return requiredSelector
+}
+
+// Create the Labels map for a Semeru Compiler
+func getLabels(wlva *wlv1.WebSphereLibertyApplication) map[string]string {
+	requiredLabels := make(map[string]string)
+	requiredLabels["app.kubernetes.io/name"] = wlva.GetName() + SemeruLabelNameSuffix
+	requiredLabels["app.kubernetes.io/instance"] = wlva.GetName() + SemeruLabelNameSuffix
+	requiredLabels["app.kubernetes.io/managed-by"] = OperatorName
+	requiredLabels["app.kubernetes.io/component"] = SemeruLabelName
+	requiredLabels["app.kubernetes.io/part-of"] = wlva.GetName()
+	return requiredLabels
+}
+
+func (r *ReconcileWebSphereLiberty) reconcileSemeruDeployment(wlva *wlv1.WebSphereLibertyApplication, deploy *appsv1.Deployment) {
+
+	deploy.Labels = getLabels(wlva)
+	deploy.Spec.Strategy.Type = appsv1.RecreateDeploymentStrategyType
+	replicas := int32(1)
+	deploy.Spec.Replicas = &replicas
+
+	if deploy.Spec.Selector == nil {
+		deploy.Spec.Selector = &metav1.LabelSelector{
+			MatchLabels: getSelectors(wlva),
+		}
+	}
+
+	// Get resources
+	semeruCloudCompiler := wlva.GetSemeruCloudCompiler()
+	instanceResources := semeruCloudCompiler.Resources
+	requestsMemory := getQuantityFromRequestsOrDefault(instanceResources, corev1.ResourceMemory, "1Gi")
```

Comment:
Good util methods to get the values. The default values have changed. Please rebase and update these values.